<a href="https://colab.research.google.com/github/eiad911/Learning-Git/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [6]:
import os
import subprocess

if not os.path.exists("ml-1st-weak"):
    subprocess.run(["git", "clone", "https://github.com/eiad911/ml-1st-weak.git"])

os.chdir("ml-1st-weak")

print(os.getcwd())

/content/ml-1st-weak


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
## My Lane

I chose the **Content Refresh Prioritization** lane.

I selected this lane because websites often contain thousands of pages, making it difficult to decide which pages should be updated first. By using data and machine learning, we can prioritize pages that are most likely to be declining and need a content refresh. This can help teams spend their time on the pages with the greatest potential impact.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
## Research Question

**Question:**
Can we identify web pages that should be prioritized for content refresh based on historical search performance and page characteristics?

**Decision:**
Which pages should be reviewed and updated first?

**Action:**
The SEO or content team will review the highest-priority pages and update their content.

**Cost of a Wrong Recommendation:**
If a declining page is missed, valuable search traffic may continue to decrease. If a healthy page is incorrectly prioritized, time and resources may be wasted.

**Why Machine Learning?**
Machine learning can combine multiple signals, such as impressions, CTR, page age, and average position, to rank pages more effectively than relying on simple manual rules.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import os
import pandas as pd

if os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("../..")

print("Current directory:", os.getcwd())

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Declining pages (%):", round((df["trend_direction"]=="down").mean()*100,2))

corr = df["search_volume"].corr(df["impressions_90d"])
print("Correlation:", round(corr,3))

Current directory: /content/ml-1st-weak
Total pages: 30000
Declining pages (%): 54.21
Correlation: 0.001


### Observations

- The dataset contains **30,000 pages**.
- About **54.2%** of the pages are labeled as declining.
- The correlation between search volume and impressions is approximately **0.001**, indicating a very weak relationship.

These observations suggest that multiple features should be considered together instead of relying on a single metric.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
## Careful Interpretation

The findings in this notebook are based only on the provided anonymized dataset. They should be interpreted as observations rather than universal conclusions. This analysis does not prove how search engines rank pages, and any future model should be evaluated carefully before being used in real decision-making.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## Self-Check

- [x] I selected a project lane.
- [x] I defined a research question.
- [x] I described the decision and action.
- [x] I explained the cost of a wrong recommendation.
- [x] I included real numbers from the dataset.
- [x] I used careful language and avoided unsupported claims.